registry/Scripts/script.sh

```diff
@@ -133,6 +136,35 @@ fetchCredentials() {
         exit $ERR_MISSING_USER_CREDENTIALS
     fi
 }
+fetchRegistryCredentials() {
+    RESOURCE=$(jq -r .authentication.audiences[0] ${ENDPOINTS} | sed "s|https://management.|https://vault.|")
+
+    TOKEN=$(curl -s --retry 5 --retry-delay 10 --max-time 60 -f -X POST \
+        -H "Content-Type: application/x-www-form-urlencoded" \
+        -d "grant_type=client_credentials" \
+        -d "client_id=${SPN_CLIENT_ID}" \
+        --data-urlencode "client_secret=${SPN_CLIENT_SECRET}" \
+        --data-urlencode "resource=${RESOURCE}" \
+        ${TOKEN_URL} | jq -r '.access_token')
+
+    KV_URL="https://${KV_NAME}.vault.${FQDN}/secrets"
+    SECRETS=$(curl -s --retry 5 --retry-delay 10 --max-time 60 -f \
+        "${KV_URL}?api-version=2016-10-01" -H "Authorization: Bearer ${TOKEN}" | jq -r .value[].id)
+
+    for secret in ${SECRETS}
+    do
+        SECRET_NAME_VERSION="${secret//$KV_URL}"
+        SECRET_NAME=$(echo ${SECRET_NAME_VERSION} | cut -d '/' -f 2)
+        SECRET_VALUE=$(curl -s --retry 5 --retry-delay 10 --max-time 60 -f \
+            "${secret}?api-version=2016-10-01" -H "Authorization: Bearer ${TOKEN}" | jq -r .value)
+        if [ $1 = "true" ]; then
+            local _REGISTRY_USER=$2
+            local _REGISTRY_PASSWORD=$3
+            eval $_REGISTRY_USER="'${SECRET_NAME}'" $_REGISTRY_PASSWORD="'${SECRET_VALUE}'"
```

Comment:
if do not think you need `eval`, this may make it visible outside the function

`REGISTRY_USER=$(echo ${SECRET_NAME_VERSION} | cut -d '/' -f 2)`